### ISSUE TITLE
After a server sleep, shm interface is lost

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#889]
// this happens when the computer is asleep, but can be reproduced by allocating a larger buffer.
// you'll have to wait for a while (it simulates sleep by buffer allocation)

```
(
fork {
    s.quit;
    1.wait;
    s.waitForBoot {
        [\hasShmInterface, s.hasShmInterface].postln; // true
        Buffer.alloc(s, 48000 * 60 * 60 * 4);
        s.sync;
        [\hasShmInterface, s.hasShmInterface].postln; // false
    }
}
)
```

// as serverRunning_(val) actually tests whether the computer is sleeping to optimise this a little,
// we need to see ifthis actually works. If not, this can be the source of the problem, but things like the above (long periods of unreachability because of a cumbersome request would still kill the shm interface.

```
(
fork {
    var t = Main.elapsedTime;
    loop {
        [Main.elapsedTime - t, thisProcess.platform.isSleeping].postln;
        0.5.wait;
    }
}
)
```

**Comment by telephon** [comment]
This is related / identical: #774

**Comment by telephon** [comment]
Since the proper refactoring (#1655) goes into 3.8, but this may be a bug in 3.7, should this fix go into 3.7.x?

**Comment by vivid-synth** [comment]
Was this fixed? The other similar issues are now closed.
